Expands docs on List rule types API (#97608)
elastic/kibana: docs/api/alerting/list_rule_types.asciidoc

<titleabbrev>List rule types</titleabbrev>
++++

Retrieve a list of alerting rule types that the user is authorized to access.

> Each rule type includes a list of consumer features within which the user is authorized to perform either `read` or `all` operations on rules of that type. This can be useful to determine which rule types a user can read but not create or modify.

Reviewer: ```suggestion
Each rule type includes a list of consumer features. Users are authorized to perform either `read` or `all` operations on rules of that type. This helps determine which rule types users can read, but not create or modify.
```

Author: Thanks Gail... my concern with this wording is that it isn't clear from that rephrasing that the user is then only authorized to read/all these rule types when they are within that list of "consumer features".
If a feature isn't on the list, then the user is not allowed to read/modify rules of that type, if they are within the missing feature.

Does that make sense?

Author: Would this work? 
```suggestion
Each rule type includes a list of consumer features. Within these features, users are authorized to perform either `read` or `all` operations on rules of that type. This helps determine which rule types users can read, but not create or modify.
```

Reviewer: I like it!

Author: Yay, thanks:)

> NOTE: Creating some rule types is limited to specific features.  These rule types are not available when <<defining-alerts, defining rules>> in <<management,Stack Management>>.

Reviewer 2: **nit**: to avoid confusion about being able to create rule types.
```suggestion
NOTE: Some rule types are limited to specific features.  These rule types are not available when <<defining-alerts, defining rules>> in <<management,Stack Management>>.
```

> NOTE: Creating some rule types may be limited to specific features, in which case they will not be available when <<defining-alerts, defining rules>> within <<management,Stack Management>>.

Reviewer: ```suggestion
NOTE: Creating some rule types is limited to specific features.  These rule types are not be available when <<defining-alerts, defining rules>> in <<management,Stack Management>>.
```
